apps/andi/lib/andi_web/live/ingestion_live_view/extract_steps/extract_http_step_form.ex

```diff
@@ -3,107 +3,101 @@ defmodule AndiWeb.ExtractSteps.ExtractHttpStepForm do
   LiveComponent for an extract step with type HTTP
   """
   use Phoenix.LiveComponent
-  import Phoenix.HTML
   import Phoenix.HTML.Form
-  import AndiWeb.Helpers.ExtractStepHelpers
+  import Phoenix.HTML
   require Logger
 
-  alias Andi.InputSchemas.ExtractSteps
   alias Andi.InputSchemas.Ingestions.ExtractHttpStep
-  alias Andi.InputSchemas.Ingestions.ExtractHeader
-  alias Andi.InputSchemas.Ingestions.ExtractQueryParam
   alias AndiWeb.EditLiveView.KeyValueEditor
   alias AndiWeb.ErrorHelpers
   alias AndiWeb.Views.Options
   alias AndiWeb.Views.DisplayNames
-  alias AndiWeb.Views.HttpStatusDescriptions
-  alias AndiWeb.Helpers.FormTools
-  alias AndiWeb.ExtractSteps.ExtractStepHeader
+  alias AndiWeb.Helpers.ExtractStepHelpers
+  alias Ecto.Changeset
+  alias Andi.InputSchemas.StructTools
+  alias AndiWeb.IngestionLiveView.ExtractSteps.ExtractStepForm
   alias Andi.UrlBuilder
-  alias Andi.InputSchemas.InputConverter
+  alias AndiWeb.Views.HttpStatusDescriptions
 
   def mount(socket) do
     {:ok,
      assign(socket,
-       testing: false,
+       testing?: false,
        test_results: nil,
        visibility: "expanded",
        validation_status: "collapsed"
      )}
   end
 
   def render(assigns) do
+    header_changesets =
+      case Changeset.fetch_change(assigns.changeset, :headers) do
```

Comment:
Are we sure these will always be a change? Would fetch_field be better? This is post validate, so I would imagine fetch_change is safe, but maybe fetch_field would be better anyways? It would allow for future changes a bit more forgivingly.